packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -107,11 +107,8 @@ export const usePostModalNavigation = (
       return null;
     }
 
-    const routeHandler = (newRoute: string) => {
-      if (newRoute.includes('/posts/')) {
-        return;
-      }
-      onCloseModal();
+    const routeHandler = () => {
+      onCloseModal(true);
```

Comment:
Even though I put true for `fromPopState` basically what it does:
- skips calling `changeHistory`
  - when this `routerHandler` is called Next.js always takes over routing to no need to worry about clearing history
- this is never called on our custom modal history changes so no need for `newRoute.includes` from my testing 
- back button now works when you are navigating between multiple tag pages by constantly selecting them eg. /tags/php -> /tags/javascript -> back -> /tags/php (before URL would change but the page would stay the same as if you are on javascript)